[EDU-3639] feat: add manage object storage guide (#576)
aziontech/docs: src/content/docs/en/pages/guides/edge-application/ea-object-storage/index.mdx

---
title: How to create and manage an object storage bucket with Azion
description: Learn how to create your buckets and upload, list, and delete files in your boject storage.
meta_tags: object storage, storage, cloud, s3, bucket, file storage
namespace: documentation_products_edge_application_object_storage
permalink: /documentation/products/guides/how-to-create-and-manage-bucket/
---

**Azion Object Storage** allows you you to create buckets and integrate your file storage with your edge infrastructure. This guide covers how to create a storage bucket, and upload, list, and delete files.

## Create and modify bucket via API

Run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and the `name` with the desired bucket name to create a new read-write bucket:

```bash

> curl --location 'https://api.azionapi.net/storage/buckets' \

Reviewer: ```suggestion
curl --location 'https://api.azion.com/v4/storage/buckets' \
```

--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{

>     "name": "my-bucket"

Reviewer: ```suggestion
    "name": "my-bucket",
```

    "edge-access": "read_write"
}'
```

## Option 1: Run cURL commands to upload and download an object from storage

1. To upload an object, run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/), `<bucket_name>` with the name of your bucket, the `Content-Type` header with the MIME type being submitted, and the file sent as a data binary:

```bash

> curl --location 'https://api.azionapi.net/storage/buckets/<bucket_name>/objects' \

Reviewer: ```suggestion
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object-name>' \
```

--header 'Accept: application/json; version=3' \
--header 'Content-Type: text/csv' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data-binary '@./path/file.csv'
```

2. To download the object, run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/), `<bucket_name>` with the name of your bucket, and `<object>` with the path, file name, and extension of the object:

```bash

> curl --location 'https://api.azionapi.net/storage/buckets/<bucket_name>/objects/<object>' \

Reviewer: ```suggestion
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object>' \
```

---
title: How to create and manage an object storage bucket

> description: Learn how to create your buckets and upload, list, and delete files in your object storage.

Author: ```suggestion
description: Learn how to create, rename, or delete your buckets from Object Storage using the Azion API.
```

meta_tags: object storage, storage, cloud, s3, bucket, file storage

> namespace: documentation_products_edge_application_object_storage

Author: ```suggestion
namespace: documentation_products_object_storage_manage
```

## Create a bucket via API

Run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) and the `name` with the desired bucket name to create a new read-write bucket:

```bash
curl --location 'https://api.azion.com/v4/storage/buckets' \
--header 'Accept: application/json; version=3' \
--header 'Content-Type: application/json' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data '{
    "name": "my-bucket",
    "edge-access": "read_write"
}'
```

## Option 1: Run cURL commands to upload and download an object from storage

1. To upload an object, run the following `POST` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/), `<bucket_name>` with the name of your bucket, the `Content-Type` header with the MIME type being submitted, and the object sent as a data binary:

```bash

> curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object-name>' \

Author: ```suggestion
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object_key>' \
```

--header 'Accept: application/json; version=3' \
--header 'Content-Type: text/csv' \
--header 'Authorization: Token [TOKEN VALUE]' \
--data-binary '@./path/file.csv'
```

2. To download the object, run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/), `<bucket_name>` with the name of your bucket, and `<object>` with the path, object name, and extension of the object:

```bash

> curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object>' \

Author: ```suggestion
curl --location 'https://api.azion.com/v4/storage/buckets/<bucket_name>/objects/<object_key>' \
```
